gh-62587: Add 'symbols' link to pydoc's html menu bar

CuriousLearner opened the pull request:
<img width="1440" alt="Screenshot 2019-05-09 at 5 09 38 PM" src="https://user-images.githubusercontent.com/8039608/57486971-54c22980-727d-11e9-9399-68166ffa16ff.png">
<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-18387](https://bugs.python.org/issue18387) -->
https://bugs.python.org/issue18387
<!-- /issue-number -->

Ref #62587 


<!-- gh-issue-number: gh-62587 -->
* Issue: gh-62587
<!-- /gh-issue-number -->

ZackerySpytz commented on Lib/pydoc.py:
```diff
@@ -430,9 +431,6 @@ def __init__(self):
         self.maxdict = 10
         self.maxstring = self.maxother = 100
 
-    def escape(self, text):
```
It seems that removing this method causes a failure in test_cgitb.

ZackerySpytz commented on Lib/pydoc.py:
```diff
@@ -2454,46 +2459,43 @@ def html_getfile(path):
 
         heading = html.heading(
-            '<big><big><strong>INDEX</strong></big></big>',
+            '<big><big><strong>%s</strong></big></big>' % title.upper(),
```
I think an f-string should be used here.

ZackerySpytz commented on Lib/pydoc.py:
```diff
@@ -2454,46 +2459,43 @@ def html_getfile(path):
+            title = 'SYMBOL'
+            contents, xrefs = htmlhelp._getsymbol(topic)
+        else:
+            raise ValueError('could not find topic %s' % repr(topic))
```
I think an f-string should be used here.

ZackerySpytz commented on Lib/pydoc.py:
```diff
@@ -2503,7 +2505,7 @@ def html_topicpage(topic):
 
             def bltinlink(name):
-                return '<a href="topic?key=%s">%s</a>' % (name, name)
+                return '<a href="topic?key=%s">%s</a>' % (html_escape(name), html_escape(name))
```
[PEP 8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) states that lines should be limited to 79 chars.

ZackerySpytz reviewed (commented):
There are several CI failures.

When creating a PR based on someone else's patch, the [developer's guide](https://devguide.python.org/pullrequest/#converting-an-existing-patch-from-b-p-o-to-github) states that the "Co-authored-by" syntax should be used in the commit message to attribute the original patch author.

CuriousLearner committed:
feat: Add 'symbols' link to pydoc's html menu bar

- Added 'symbols' navigation link alongside 'topics' and 'keywords'
- Refactored html_topicindex() to handle topics, keywords, and symbols
- Added _getsymbol() method to Helper class for symbol lookup
- Restored escape() method in HTMLRepr and HTMLDoc classes
- Updated test suite to include symbols link verification
- Modernized string formatting using f-strings

Original patch by Ron Adam.

Co-authored-by: Ron Adam <ron3200@gmail.com>

CuriousLearner committed:
Merge branch 'main' into fix-Issue18387

auvipy reviewed (approved):
looks better

CuriousLearner committed:
Merge branch 'main' into fix-Issue18387
